TLM/TLM/Manager/Impl/RoutingManager.cs

```diff
@@ -536,196 +578,199 @@ public class RoutingManager : AbstractGeometryObservingManager, IRoutingManager
 										if (debugFine)
 											Log._Debug($"RoutingManager.RecalculateLaneEndRoutingData({segmentId}, {laneIndex}, {laneId}, {startNode}): lane arrow check FAILED for nextLaneId={nextLaneId}, idx={nextLaneIndex}. adding as relaxed lane.");
 #endif
 
-										// lane can be used by all vehicles that may disregard lane arrows
-										transitionType = LaneEndTransitionType.Relaxed;
-										if (numNextRelaxedTransitionDatas < MAX_NUM_TRANSITIONS) {
-											nextRelaxedTransitionDatas[numNextRelaxedTransitionDatas++].Set(nextLaneId, nextLaneIndex, transitionType, nextSegmentId, isNextStartNodeOfNextSegment, GlobalConfig.Instance.PathFinding.IncompatibleLaneDistance);
-										} else {
-											Log.Warning($"nextTransitionDatas overflow @ source lane {prevLaneId}, idx {prevLaneIndex} @ seg. {prevSegmentId}");
-										}
-									}
-								} else if (!nextHasOutgoingConnections) {
+                                        // lane can be used by all vehicles that may disregard lane arrows
+                                        transitionType = LaneEndTransitionType.Relaxed;
+                                        if (numNextRelaxedTransitionDatas < MAX_NUM_TRANSITIONS) {
+                                            if (nextIsJunction || laneInValidGroup)
+                                                nextRelaxedTransitionDatas[numNextRelaxedTransitionDatas++].Set(nextLaneId, nextLaneIndex, transitionType, nextSegmentId, isNextStartNodeOfNextSegment, GlobalConfig.Instance.PathFinding.IncompatibleLaneDistance);
+                                        } else {
+                                            Log.Warning($"nextTransitionDatas overflow @ source lane {prevLaneId}, idx {prevLaneIndex} @ seg. {prevSegmentId}");
+                                        }
+                                    }
+                                } else if (!nextHasOutgoingConnections) {
 #if DEBUGROUTING
 									if (debugFine)
 										Log._Debug($"RoutingManager.RecalculateLaneEndRoutingData({segmentId}, {laneIndex}, {laneId}, {startNode}): nextLaneId={nextLaneId}, idx={nextLaneIndex} is used by vehicles that do not follow lane arrows. adding as default.");
 #endif
 
-									// routed vehicle that does not follow lane arrows (trains, trams, metros, monorails)
-									transitionType = LaneEndTransitionType.Default;
+                                    // routed vehicle that does not follow lane arrows (trains, trams, metros, monorails)
+                                    transitionType = LaneEndTransitionType.Default;
 
-									if (numNextForcedTransitionDatas < MAX_NUM_TRANSITIONS) {
-										nextForcedTransitionDatas[numNextForcedTransitionDatas].Set(nextLaneId, nextLaneIndex, transitionType, nextSegmentId, isNextStartNodeOfNextSegment);
+                                    if (numNextForcedTransitionDatas < MAX_NUM_TRANSITIONS) {
+                                        if (nextIsJunction || laneInValidGroup)
```

Comment:
don't know, might pose a problem. Do trains need to care for medians?

Reply:
As long as trains are encouraged to remain on their same track (as implemented in v10.17) there shouldn't be a problem? Except at ends of stations that have 'bypass' tracks?